tensorflow/compiler/xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -2174,15 +2174,19 @@ void IrEmitterUnnested::EmitPrologueForReduction(
     llvm::Type* element_type =
         llvm_ir::PrimitiveTypeToIrType(reduce_inst->shape().element_type(),
                                        ir_emitter_context_->llvm_module());
-    llvm::AllocaInst* reduction_input_address = Alloca(element_type);
+    llvm::AllocaInst* reduction_input_address =
+        llvm_ir::EmitAllocaAtFunctionEntry(element_type, absl::string_view(),
+                                           &b_);
     reduction_input_addresses->push_back(reduction_input_address);
 
     int num_partial_results = reduction_info->GetNumPartialResults();
     AddressVector* partial_result_addresses =
         reduction_info->GetMutablePartialResultAddresses();
+    std::string alloca_name = "partial_reduction_result." + std::to_string(i);
```

Comment:
try use `llvm::Twine` here for string manipulation.

Reply:
and pass it to `absll:string_view` later on with `Twine::str()` member function.

Reply:
I had left that call as is, in the first iteration...but it gave a compile error...did not try to dive into it too much, since the fix was trivial

Reply:
in XLA and LLVM, the basic rule is to avoid `std::string` as much as possible, especially in the internal routine. `std::string` is normally only used interfacing with external worlds such as parsing command line options.